site/docs/develop/spec.md

```diff
@@ -1,80 +1,180 @@
+1. the pointer to the Iceberg _table metadata_ and
+2. the ID of the _Iceberg snapshot_ within the Iceberg _table metadata_.
+
+The pointer to the Iceberg table is recorded as _Global State_ and the ID of the Iceberg snapshot
+is recorded within the _Put operation_ in a Nessie commit.
+
+!!! note
+    This model puts a strong restriction on the Iceberg table. All metadata JSON documents must be
+    stored and none of the built-in iceberg maintenance procedures can be used. There are
+    potentially serious issues regarding schema migrations in this model as well. Therefore, the
+    Iceberg table spec should be considered subject to change in the near future.
+
+#### Delta Lake Table
 
-The reason for this condition is to behave like a ‘real’ database. You shouldn’t have to update your reference before transacting on tableA because it just happened to update tableB whilst you were preparing your transaction.
+The state of a Delta Lake Table is represented using the Delta Lake Table attributes
+`metadataLocationHistory`, `checkpointLocationHistory` and `lastCheckpoint`.
 
-## Tiered Version Store
+Delta Lake Tables are tracked without a _Global State_ in Nessie, i.e. those three attributes are
+recorded within the [_Put Operation_](#put-operation) of a Nessie commit.
+
+#### View
+
+The state of an SQL view is represented using the attributes
+`sqlText` and `dialect` (currently one of `HIVE`, `SPARK`, `DREMIO`, `PRESTO`).
+
+SQL views are tracked without a _Global State_ in Nessie, i.e. those three attributes are
+recorded within the [_Put Operation_](#put-operation) of a Nessie commit.
+
+## Operations in a Nessie commit
+
+Each Nessie commit carries one or more operations. Each operation contains the Content Key plus and
+comes in one of the following variations.
+
+### Put operation
+
+A _Put operation_ modifies the state of the included Content object. It must contain the Content
+object and, if the Content type tracks [_Global State_](#on-reference-state-vs-global-state), also
+the _expected contents_. The _expected contents_ attribute can be omitted, if the Content object
+refers to a new Content Id, e.g. a newly created table or view.
+
+### Delete operation
+
+A _Delete operation_ does not carry any Content object and is used to indicate that a Content
+object is no longer referenced using the Content Key of the _Delete operation_.
+
+### Unmodified operation
+
+An _Unmodified operation_ does not represent any change of the data, but can be included in a
+Nessie commit operation to enforce strict serializable transactions. The presence of an
+_Unmodified operation_ means that the Content object referred to via the operation's Content Key
+must not have been modified since the Nessie commit's `expectedHash`.
+
```

Comment:
Will it be good if we add here about how each of Nessie operations(from CLI / API / SQL) mapped to these commit operations for easy understanding and reference?

For example, create branch maps to which above commit operation  and which Nessie operation leads to Unmodified operation  ?

Reply:
Added a bit at the description of the operations.

Reply:
ok. Fine as of now. Later if required I will add a table mentioning how each nessie CLI operations is mapped to commit operations with content of each.